lib/metrics-tracker.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const { Labels } = require('./constants');
+
+class MetricsTracker {
+    constructor({ metrics }) {
+        this.metrics = metrics;
+    }
+
+    async trackHistogramDuration({ metricName, labels, action, handleResult }) {
+        if (!this.metrics) {
+            return action();
```

Comment:
Potential `null` reference here. Some validation for the required arguments would be nice

Reply:
Done